internal/npm_install/generate_build_file.js

```diff
@@ -745,8 +748,11 @@ function flattenPkgDependencies(pkg, dep, pkgsMap) {
     });
   }
 
-  findDeps(dep.dependencies, true, 'dependency');
-  findDeps(dep.peerDependencies, true, 'peer dependency');
+  findDeps(dep.dependencies, () => true, 'dependency');
+  findDeps(dep.peerDependencies, (targetDep) => {
+    // if it is not in the can ignore then is is required
+    return canIgnore.indexOf(targetDep) === -1;
```

Comment:
so the field `canIgnore` is a list in the user's package.json? a new field you're adding?

Would be good to have some doc change so users can figure out how to use this, along with some test (can just be that you find a way to reproduce the failure in this repo by not installing a peer dep, then use your new feature to fix that breakage)

Reply:
Yes, `canIgnore` is a string list in the package.json, I was going to add documentation but I don't know whether or not to do it this way, we also have an alternative of putting in a rule attribute, which would you prefer?

Reply:
I'd prefer an attribute over a new key added to the package.json

Reply:
@Toxicable I personally think I am with you, it was something I thought about after I wrote this but will amend asap.

Reply:
yeah it's a behavior of `npm_install`/`yarn_install` so seems like it should go there. I think the default should be to act like `npm` and `yarn` which means we should print peerDep warnings but proceed. Offer users a `strictPeerDependencies` boolean option. (I guess you could have a list of specific peerDep warnings to whitelist in strict mode)

Reply:
I have changed the way it is done, will check when at work again tomorrow to make sure it still works if so I will add documentation if it is ok with you?